elyra/pipeline/local/processor_local.py

```diff
@@ -114,6 +115,18 @@ def process(self, pipeline):
     def export(self, pipeline, pipeline_export_format, pipeline_export_path, overwrite):
         raise NotImplementedError("Local pipelines does not support export functionality")
 
+    @property
+    def supported_properties(self) -> List[str]:
+        """A list of Elyra-owned properties supported by this runtime processor."""
+        return [
+            pipeline_constants.ENV_VARIABLES,
+            pipeline_constants.KUBERNETES_SECRETS,
+            pipeline_constants.MOUNTED_VOLUMES,
+            pipeline_constants.KUBERNETES_POD_ANNOTATIONS,
+            pipeline_constants.KUBERNETES_TOLERATIONS,
+            pipeline_constants.DISABLE_NODE_CACHING,
+        ]
```

Comment:
I think only env variables are support by the local processor.  I'm guessing this is another example of not promoting local _mode_ to be its own _processor type_, and what is really meant by this list are the properties that can be associated with Notebook, Python, and R nodes.  The `LocalPipelineProcessor` (today) only knows how to deal with envs.

Reply:
Yes, that's exactly it. I had this in the PipelineProcessor before, perhaps I should move it back there for clarity?

Reply:
I see, thanks.  I think it makes sense to leave here, but only contain `env_vars`, since others don't apply to `LocalPipelineProcessor`.

I suppose you could move to `PipelineProcessor` but only define env_vars there (since, I'm guessing, every runtime will support env variables).  Then you wouldn't need the attribute check.  If we found that some new processor didn't support envs, then we could have the base class return the empty array and move the envs_vars to `LocalPipelineProcessor` at that time - which still removes the need for the attribute check.

Reply:
Well the issue is that the call to grab the "local" properties has to display all the properties available for generic nodes in any runtime type since they can be run anywhere. That's actually why, to determine the properties to render for each runtime, I'm using this "supported_properties" method instead of the class attribute check for an "add_{property}" method. Since "local" needs to pull in all these properties for which an "add" function will never be relevant, this list seemed a slightly better alternative. And, as we chatted about, we can't use the parent class to do the "add_{property}" attribute check since then there will never be a difference in properties available between runtimes.

Reply:
I think I could probably do a thing where, when "local" properties are being queried, take the set of properties supported by all subclasses of 'PipelineProcessor' (or 'RuntimePipelineProcessor'). For reasons too long to explain here, this wouldn't require an explicit reference to "local" in the code

Reply:
Thank you for the explanations.  This isn't a big enough deal to spend time on at the moment.  It's helpful to know that this is a side-effect of _not_ making "local mode" its own runtime and something we can address should we decide to promote it to its own runtime down the road.  Thanks for spending time on this.  I know things like this can be extremely frustrating.